lib/src/box.dart

```diff
@@ -166,5 +166,29 @@ class Box<T> {
     return _getMany(() => checkObxPtr(bindings.obx_box_get_all(_cBox), "failed to get all objects from box", true));
   }
 
+  // same as calling maxCount with limit := 0
+  int count() {
+    return maxCount(limit: 0);
+  }
+
+  int maxCount({int limit}) {
+    Pointer<Uint64> _count = Pointer<Uint64>.allocate();
+    checkObx(bindings.obx_box_count(_cBox, limit, _count));
+    return _count.load<int>();
+  }
```

Comment:
Having two methods is not necessary. I assume you've been inspired by the Go implementation, however, it needs to use two methods because there are no optional arguments in Go.

In dart, we can have just `count({int limit})`

Reply:
Ah yeah of course!

Reply:
Do you think a named parameter or a true optional `[limit = 0]` param should be used? As the docs mention .count() can be called with an optional limit -> which implies the parameter doesn't need to be named.

Reply:
I think named parameter would be good in this case